libs/platform/core/src/services/app-initializer/app-initializer.ts

```diff
@@ -0,0 +1,20 @@
+import { OnDestroy } from '@spryker-oryx/di';
+import { Observable } from 'rxjs';
+
+export const AppInitializerService = 'oryx.AppInitializerService';
+export const AppInitializer = 'oryx.AppInitializer*';
+
+export interface AppInitializerService extends OnDestroy {
+  initialize(): void;
+}
+
+export interface AppInitializer {
+  initialize(): void | Observable<unknown>;
```

Comment:
Return type should be consistent, also maybe we can allow for sync methods as well:
```suggestion
  initialize(): void | Promise<void> | Observable<void>;
```